Update metrics docs for JS (#3853)
open-telemetry/opentelemetry.io: content/en/docs/languages/js/instrumentation.md

## Metrics

> [Metrics](/docs/concepts/signals/metrics) combine individual measurements into
> aggregations, and produce data which is constant as a function of system load.
> The aggregations lack details required to diagnose low level issues, but

Reviewer: "aggregations" --> "aggregates"?

```suggestion
[Metrics](/docs/concepts/signals/metrics) combine individual measurements into
aggregates, and produce data which is constant as a function of system load.
Aggregates lack details required to diagnose low level issues, but
```

complement spans by helping to identify trends and providing application runtime
telemetry.

The metrics API defines a variety of instruments. Instruments record
measurements, which are aggregated by the metrics SDK and eventually exported
out of process. Instruments come in synchronous and asynchronous varieties.

> Synchronous instruments record measurements as they happen. Asynchronous
> instrument register a callback, which is invoked once per collection, and which
> records measurements at that point in time.

Reviewer: ```suggestion
Synchronous instruments record measurements as they happen. Asynchronous
instruments register a callback that is invoked once per collection, and which
record measurements at that point in time.
```

"at that point in time" isn't as clear as it could be, but it's good enough for now.

suited for your use case, see
[Supplementary Guidelines](/docs/specs/otel/metrics/supplementary-guidelines/).

{%% alert title="Note" class="info" %%} OpenTelemetry instruments are either
synchronous or asynchronous (observable).

> If you want to learn more about the difference, and when to use which kind, take
> a look at the concept page on

Reviewer 2: ```suggestion
To learn more about the differences between synchronous and
asynchronous instruments, and when to use each kind, see the
concept page on
```

suited for your use case, see

> [Supplementary Guidelines](/docs/specs/otel/metrics/supplementary-guidelines/).

Reviewer: Could we consolidate this sentence with the alert you added just below. It doesn't make sense to have a normal paragraph refer the reader to the spec via: "For more on synchronous and asynchronous instruments, and which kind is best suited for your use case, see [Supplementary Guidelines](/docs/specs/otel/metrics/supplementary-guidelines/)" immediately followed by an alert that says ~ "for more info see the concept page".

Author: ah, oh, I didn't see that. This may also fix some of the things above.

> {%% alert title="Note" class="info" %%} OpenTelemetry instruments are either

Reviewer: ```suggestion
{{% alert title="Note" class="info" %}} OpenTelemetry instruments are either
```

synchronous or asynchronous (observable).

To learn more about the differences between synchronous and
asynchronous instruments, and when to use each kind, see the
concept page on
[synchronous and asynchronous instruments](/docs/concepts/signals/metrics/#synchronous-and-asynchronous-instruments)

> {%% /alert %%}

Reviewer: ```suggestion
{{% /alert %}}
```

> {{% alert title="Note" class="info" %}} OpenTelemetry instruments are either
> synchronous or asynchronous (observable).
> 
> To learn more about the differences between synchronous and
> asynchronous instruments, and when to use each kind, see the
> concept page on
> [synchronous and asynchronous instruments](/docs/concepts/signals/metrics/#synchronous-and-asynchronous-instruments)
> {{% /alert %}}

Author: ```suggestion
```

const metricReader = new PeriodicExportingMetricReader({
  exporter: new ConsoleMetricExporter(),

>   // Default is 60000ms (60 seconds). Set to 5 seconds for demonstrative purposes only.
>   exportIntervalMillis: 5000,

Reviewer: A general comment since it has been coming up lately: to me it's a bad smell that we're repeating the default value in several places in the docs. This will be a pain to update if the default happens to change. How about something like:

```suggestion
  exportIntervalMillis: 5000, // Reduce export interval for demo purposes only.
```

I'd rather that we link to a page that says what the default is instead of repeating the default value.

Reviewer: Also, I think that we agreed elsewhere that this would be ~ 10 sec?

Author: > in the PR for java who discusses to introduce that

Also, I think that we agreed elsewhere that this would be ~ 10 sec?

Author: let's have a dedicated issue to make sure that we have some consistency here.
